### PR TITLE
FIX: Check function attributes upon class creation

### LINKED ISSUE #411: Using a function as a default for a Callable in a child class fails
I have a `HasTraits` class with some `Callable` traits, and a second class that derives from it. I was trying to set the defaults for these traits as class variables, but I noticed that sometimes these defaults didn't get applied. The following example will make this clear. Here I have two Callables in the base class, and one is initialized in the derived class with a builtin function and one with a regular function. The latter does not get applied.

```python
from traits.api import HasStrictTraits, Callable

def regular_fun():
    pass

builtin_fun = print

class A(HasStrictTraits):
    f1 = Callable()
    f2 = Callable()

class B(A):
    f1 = builtin_fun
    f2 = regular_fun

if __name__ == '__main__':
    b = B()
    assert b.f1 is not None
    assert b.f2 is not None
```

Under the tested version (4.6.0) the first assert passes (as expected), but the second assert fails (`b.f2` is None). 

This is because in `MetaHasTraitsObject` we check whether items in the class_dict are functions: https://github.com/enthought/traits/blob/b5cf152c446bb87d575b5ecdc4cda7f8b646d96f/traits/has_traits.py#L550-L553

Builtin functions don't pass this check, but regular functions do. So `f2` gets treated in this block and ultimately ignored, but `f1` skips this block and ends up in the `else` where we look at inherited class traits:

https://github.com/enthought/traits/blob/b5cf152c446bb87d575b5ecdc4cda7f8b646d96f/traits/has_traits.py#L577-L590

Here, `builtin_fun` gets correctly set as the default for `f1`, but `f2` has already been treated and therefore keeps it None default.

### PR DESCRIPTION
**Pull request by jvkersch**
fixes #411

### CODE REVIEW

File: traits/tests/test_regression.py [hunk 1]
```diff
@@ -232,5 +234,28 @@ def test_on_trait_change_with_list_of_extended_names(self):
...
+            f = Callable()
+
+        class B(A):
+            f = fun
+
+        # When
+        b = B()
+
+        # Then
+        self.assertIsNotNone(b.f)
```
Comment (mdickinson) [inline traits/tests/test_regression.py]: Could we go one step further and call `b.f`, and verify that the result is what we expect (`None` in this case, but we could fix up `fun` to return something interesting instead).

**Comment by jvkersch** [comment]
Done. Thanks for pointing that out!

**Comment by mdickinson** [comment]
LGTM. Thanks! (Waiting for CI before merging.)

**Comment by rkern** [comment]
This seems to be causing problems and should probably be reverted until they can be worked out: https://github.com/enthought/codetools/pull/22#issuecomment-434166592

**Comment by rkern** [comment]
Minimal test case that fails on `master` and succeeds after reverting this PR:

```
import unittest

from traits.api import HasTraits


class Foo(HasTraits):
    pass


class TestHasAttrMethod(unittest.TestCase):

    def test_subclass(self):
        foo = Foo()
        self.assertFalse(hasattr(foo, 'method'))

        class Bar(Foo):
            def method(self):
                return True

        bar = Bar()
        self.assertTrue(hasattr(bar, 'method'))
        self.assertTrue(bar.method())
```

**Comment by mdickinson** [comment]
Thanks, @rkern. Reverting and re-opening the original issue.

**Pull request merged**